eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -210,6 +210,16 @@ jobs:
     displayName: Publish Code Coverage to DevOps
     condition: eq(variables['AZ_SDK_CODE_COV'], 1)
 
+  - task: mspremier.BuildQualityChecks.QualityChecks-task.BuildQualityChecks@6
+    displayName: Check coverage
+    inputs:
+      checkCoverage: true
+      coverageFailOption: fixed
+      coverageType: line
+      # 85% minimum line coverage
```

Comment:
This will be 85% of changed lines or all lines?

Reply:
I think it's "total coverable lines" based on the log output: https://dev.azure.com/azure-sdk/public/_build/results?buildId=339456&view=logs&j=c405da93-63bc-52c4-660e-dc11361a3bc8&t=c4c64a0a-81b3-5829-a7b2-c5e677c70404&l=14

```
Total line: 1341
Covered line: 1246
Code Coverage (%): 92.9157
```

"Coverable lines" in the coverage report is 1341-- https://dev.azure.com/azure-sdk/public/_build/results?buildId=339456&view=codecoverage-tab

![image](https://user-images.githubusercontent.com/2158838/78083618-dece2180-736a-11ea-821b-93e23f2263c3.png)

Reply:
@vhvb1989 -- 1341 seems like a low number for coverable lines compared to the total of 5,155 lines. Is the coverage tool configured properly?

Reply:
I did a quick check on some of the files like `az_json_builder_c` that make up the set of files being measured here, and the numbers make sense. `az_json_builder_c` has 168 total lines (include whitespace/comments) but ~80 lines of actual code.

That said, the header files, don't get accounted for accurately (including macros and typedefs).

For example, most of `az_span_h` is "grey":
https://dev.azure.com/azure-sdk/29ec6040-b234-4e31-b139-33dc4287b756/_apis/test/CodeCoverage/browse/5246113/Code%20Coverage%20Report_339456/inc_az_span_h.htm

And these methods in `az_http_internal_h` look like they should certainly be accounted for and considered either "covered" or needs to be covered (and not grey):
![image](https://user-images.githubusercontent.com/6527137/78176854-0926ea00-7412-11ea-9f02-3beba20bfb39.png)

We also seem to only be including Azure Core and not other components like IOT/etc. Which is fine for now, but just wondering if that was intentional.

Reply:
Yeah, this looks right.

Reply:
What about those `az_http_internal_h` methods, @vhvb1989?

Reply:
@ahsonkhan that depends on the compiler.
even though we are building debug, compiler will take all INLINE functions and optimize the code inside any function calling them.
We would need to sacrifice all INLINE functions to see coverage.
Or we can manually make sure that these INLINE functions are used from tests cases.

And the other gray lines we see are only functions definitions. We can't see coverage for those.

We should be generating coverage for iot as well.
I added this for storage, keyvault, iot and core.
Provably it got removed after their test refactor (@danewalton FYI), we might want to create an issue to fix that on iot.

on tests projects we use
```cmake
include(AddTestCMocka)
#and then
add_cmocka_test(${TARGET_NAME} SOURCES..)
# this generates a %{TARGET_NAME}_cov that can be run to generate covergare
```

Reply:
the issue is actually not because their refactor, but because the name of their project change and CI is still trying to call the previous name. 
```yml
  - script: |
      make az_core_cov_xml
      make az_iot_cov_xml
      make az_keyvault_cov_xml
      make az_storage_blobs_cov_xml
```

We just need to update the name and add any other iot project. 
I will mention this on the channel